adagrams/game.py

```diff
@@ -1,11 +1,96 @@
+import random
+import copy
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+        1: ["A", "E", "I", "O", "U", "N", "R", "S", "T"],
+        2: ["D", "G"],
+        3: ["B", "C", "M", "P"],
+        4: ["F", "H", "V", "W", "Y"],
+        5: ["K"],
+        8: ["J", "X"],
+        10: ["Q", "Z"]
+    }
+
+
+
 def draw_letters():
-    pass
+    letters_for_user = []
+    fresh_copy = copy.copy(LETTER_POOL)
+    while len(letters_for_user) < 10:
+        letter = random.choice(list(fresh_copy.keys()))
+        value = fresh_copy[letter]
+        if value > 0:
+            letters_for_user.append(letter)
+            fresh_copy[letter] -= 1
+    return letters_for_user
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_copy = copy.copy(letter_bank)
+    word = word.upper()
+    for letter in word:
+        if letter in letter_bank_copy:
+            letter_bank_copy.remove(letter)
+        else:
+            return False
+    return True
+
+
 
 def score_word(word):
-    pass
+    score = 0
+    word = word.upper()
+    for letter in word: 
+        for key, value in SCORE_CHART.items():
+            if letter in value:
```

Comment:
A thought here about the structure of SCORE_CHART: I do think that having the  score be the key and having the list of letters as the value makes this data structure look more concise because it takes up fewer lines of code. 

However, when you iterate through the dict on line 73 and then check `if letter in value` on line 74, you already have 2 nested for loops and then the `in` operator checking a list. So your worse case scenario is that you have to iterate through every letter in word, iterate through every key/value pair in the dict, and then iterate through ever letter in value (which is a list) -- quickly growing your time complexity. 

If SCORE_CHART was a dict with 26 key value pairs where each letter was the key and each value was the score, then you could leverage a dictionary's constant look up time.